### PR TITLE
feat(repo): use better xcodebuild logging for updates-e2e test

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- xcodebuild outputs a ridiculous amount of logging by default, this makes it hard to inspect what went wrong in GitHub console.

# How

- use `@expo/xcpretty` to format logs the same way we do in Expo CLI and EAS Build.
- use `create-expo-app` instead of `expo init`.
- make it a bit easier to re-run the test locally.

# Test Plan

- runs locally, will see how it does in CI

### CODE REVIEW

**Review by wschurman — APPROVED** [review]

File: packages/expo-updates/e2e/__tests__/setup/project.js [hunk 1]
```diff
@@ -166,16 +170,24 @@ async function initAsync(workingDir, repoRoot, runtimeVersion) {
...
-  );
+  const localTemplatePathName = glob.sync(
+    path.join(projectRoot, 'expo-template-bare-minimum-*.tgz')
+  )[0];
+
+  if (!localTemplatePathName) {
+    throw new Error(`Failed to locate packed template in ${projectRoot}`);
+  }
+
+  await spawnAsync(localCliBin, ['prebuild', '--template', localTemplatePathName], {
```
Comment (douglowder) [inline packages/expo-updates/e2e/__tests__/setup/project.js:182]:
```suggestion
  await spawnAsync('npx', ['expo', 'prebuild', '--template', localTemplatePathName], {
```

**Review by douglowder — CHANGES_REQUESTED** [review]
I'm still not able to complete the setup script locally with these changes. I did have one code suggestion that at least for me gets past prebuild -- still seeing pod install fail, but that might be just my local setup. I'll keep investigating.

File: packages/expo-updates/e2e/__tests__/setup/project.js (hunk 1 above)
Comment (douglowder) [reply packages/expo-updates/e2e/__tests__/setup/project.js:182]: Actually cancel this code suggestion, I figured out what the real problem is. The old global Expo CLI is still being used in several places in the setup.

**Review by douglowder — APPROVED** [review]

File: packages/expo-updates/e2e/__tests__/setup/project.js (hunk 1 above)
Comment (douglowder) [reply packages/expo-updates/e2e/__tests__/setup/project.js:180]: Ok @EvanBacon it looks like there were a few unrelated issues in the E2E flow :) I've added a few more commits and it is now working, so you can go ahead and merge. Thanks very much for this, the formatted logs are a lot better!

**Pull request merged**